src/elements/element.rectangle.js

```diff
@@ -100,45 +197,16 @@ module.exports = function(Chart) {
 				}
 			}
 
-			ctx.beginPath();
-			ctx.fillStyle = vm.backgroundColor;
-			ctx.strokeStyle = vm.borderColor;
-			ctx.lineWidth = borderWidth;
-
-			// Corner points, from bottom-left to bottom-right clockwise
-			// | 1 2 |
-			// | 0 3 |
-			var corners = [
-				[left, bottom],
-				[left, top],
-				[right, top],
-				[right, bottom]
-			];
-
-			// Find first (starting) corner with fallback to 'bottom'
-			var borders = ['bottom', 'left', 'top', 'right'];
-			var startCorner = borders.indexOf(borderSkipped, 0);
-			if (startCorner === -1) {
-				startCorner = 0;
-			}
-
-			function cornerAt(index) {
-				return corners[(startCorner + index) % 4];
-			}
-
-			// Draw rectangle from 'startCorner'
-			var corner = cornerAt(0);
-			ctx.moveTo(corner[0], corner[1]);
-
-			for (var i = 1; i < 4; i++) {
-				corner = cornerAt(i);
-				ctx.lineTo(corner[0], corner[1]);
-			}
+			var borders = {
```

Comment:
For performance sake, this **internal** borders object should be an array of array (properly commented):

```javascript
var borders = [
    [x0, y0, x1, y1],  //< top
    [x0, y0, x1, y1],  //< right
    [x0, y0, x1, y1],  //< bottom
    [x0, y0, x1, y1],  //< left
];
```